Consigment/ConsigmentTests/UnitTest1.cs

```diff
@@ -0,0 +1,121 @@
+using ConsigmentStruct;
+
+namespace ConsigmentTests
+{
+    [TestFixture]
+    public class ConsigmentTests
+    {
+        [Test]
+        public void ConstructorTest()
+        {
+            var consigment = new Consigment(21, 18.5);
+            Assert.That(consigment.Quantity, Is.EqualTo(21));
+            Assert.That(consigment.Price, Is.EqualTo(18.5));
+        }
+        [TestCase(-30)]
+        public void QuantitySet_NegativeValue_ArgumentException(int val)
+        {
+            var consigment = new Consigment();
+            Assert.That(() => consigment.Quantity = val, Throws.ArgumentException);
+        }
+        [TestCase(-30.99)]
+        public void PriceSet_NegativeValue_ArgumentException(double val)
+        {
+            var consigment = new Consigment();
+            Assert.That(() => consigment.Price = val, Throws.ArgumentException);
+        }
+        [TestCase(15, 42.47, 637.05)]
+        [TestCase(0, 0, 0)]
+        public void CostTest(int quantity, double price, double result)
+        {
+            var consgiment = new Consigment(quantity, price);
+            Assert.That(consgiment.Cost, Is.EqualTo(result));
+        }
+        [TestCase(15, 42.47, "15 ��. �� 42,47 ���.\"")]
+        [TestCase(0, 0, "0 ��. �� 0 ���.\"")]
+        [TestCase(42, 18, "42 ��. �� 18 ���.\"")]
+        public void ToStringTest(int quantity, double price, string result)
+        {
+            var consigment = new Consigment(quantity, price);
+            Assert.That(consigment.ToString(), Is.EqualTo(result));
+        }
+        [TestCase(30, 30, 30, 30, true)]
+        [TestCase(30, 15, 30, 30, false)]
+        public void Equals_TwoConsigments_ExpectedResult(int quantity1, double price1, int quantity2, double price2, bool result)
+        {
+            var consigment1 = new Consigment(quantity1, price1);
+            var consigment2 = new Consigment(quantity2, price2);
+            Assert.That(consigment1.Equals(consigment2), Is.EqualTo(result));
+        }
+        [Test]
+        public void Equals_WrongArgument_ArgumentException()
+        {
+            var consigment = new Consigment();
+            var smth = new object();
+            Assert.That(() => consigment.Equals(smth), Throws.ArgumentException);
+        }
+        [Test]
+        public static void GetHashCodeTest()
+        {
+            var x = new Consigment(45, 18);
+            var y = new Consigment(45, 18);
+            var z = new Consigment(40, 45);
+            Assert.That(x.Equals(y), Is.True);
+            Assert.That(x.Equals(z), Is.False);
+        }
```

Comment:
Этот тест лучше было сделать параметрическим и протестировать несколько случаев. Например, так
```suggestion
        [TestCase(45, 18, true)]
        [TestCase(40, 45, false)]
        [TestCase(45, 20, false)]
        [TestCase(40, 18, false)]
        [TestCase(30, 27, false)]
        public static void GetHashCodeTest(int quantity, double price, bool expected)
        {
            var x = new Consigment(45, 18);
            var y = new Consigment(quantity, price);

            var actual = x.GetHashCode() == y.GetHashCode();

            Assert.That(actual, Is.EqualTo(expected));
        }
```
Заметьте, что в последнем случае для вашей реализации `GetHashCode()` тест не проходит.